### PR TITLE
Adding Python panel examples

### PR DESCRIPTION
**Pull request by Br2850**
What's changed?

- [x] Added README.md for Python panel examples following the existing `plugins/` paradigm
- [x] Added examples & documented py panel specific resources used for:
   - [x] Counter
   - [x] Plot
   - [x] Markdown
   - [x] Table
   - [x] Media Player
   - [x] Image
   - [x] Multiple Views in 1 Panel
   - [x] Accepting User Input
   - [x] Interactive Plot
   - [x] Dropdown Menu
   - [x] Walkthrough

### CODE REVIEW

**Comment by Br2850** [comment]
Replaces: https://github.com/voxel51/fiftyone-plugins/pull/139

**Review by ritch — APPROVED** [review]

File: plugins/panel-examples/__init__.py [hunk 1]
```diff
@@ -0,0 +1,940 @@
...
+    @property
+    def config(self):
+        return foo.PanelOperatorConfig(
+            name="example_plot", label="Python Panel Example: Plot"
+        )
+
+    @staticmethod
+    def on_load(ctx: ExecutionContext):
+        # set plot data with object
+        plot_data = {
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:84]: Plots should use data, not state

File: plugins/panel-examples/__init__.py [hunk 2]
```diff
@@ -0,0 +1,940 @@
...
+
+    @staticmethod
+    def on_load(ctx: ExecutionContext):
+        ctx.panel.state.table = [
+            {"name": "John", "age": 30},
+            {"name": "Jane", "age": 32},
+        ]
+
+        ctx.panel.state.code = "def main():\n\tprint('Hello World!')"
+        ctx.panel.state.plot = [
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:392]: plot should use data, not state

File: plugins/panel-examples/__init__.py [hunk 3]
```diff
@@ -0,0 +1,940 @@
...
+
+###
+# Counter
+###
+
+
+class CounterPanel(foo.Panel):
+    @property
+    def config(self):
+        return foo.PanelOperatorConfig(
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:21]: `PanelConfig`

File: plugins/panel-examples/__init__.py [hunk 4]
```diff
@@ -0,0 +1,940 @@
...
+            name="example_interactive_plot",
+            label="Python Panel Example: Interactive Plot",
+        )
+
+    @staticmethod
+    def on_load(ctx: ExecutionContext):
+
+        # tabulate histogram values
+        label_counts = {}
+        for sample in ctx.dataset.iter_samples():
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:559]: Not efficient, should be updated to use `dataset.histogram_values()` or similar

File: plugins/panel-examples/__init__.py [hunk 5]
```diff
@@ -0,0 +1,940 @@
...
+            }
+        ]
+        layout = {
+            "width": 600,
+            "xaxis": {"title": "Label Name"},
+            "yaxis": {"title": "Count"},
+            "title": "A Fancy Plot",
+        }
+
+        ctx.panel.state.histogram = histogram_data
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:593]: should use data, not state

File: plugins/panel-examples/__init__.py [hunk 6]
```diff
@@ -0,0 +1,940 @@
...
+        pass
+
+    def filter_data(self, ctx: ExecutionContext):
+        filter_label = ctx.params["data"]["label"]
+        # create a view of the dataset that only includes samples with the filter label
+        filtered_view = ctx.dataset.filter_labels(
+            "ground_truth", F("label") == filter_label
+        )
+        # display the filtered view
+        ctx.trigger(
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:613]: Use `ctx.ops.set_view()`

File: plugins/panel-examples/__init__.py [hunk 7]
```diff
@@ -0,0 +1,940 @@
...
+            "set_view",
+            params=dict(
+                view=json.loads(json_util.dumps(filtered_view._serialize()))
+            ),
+        )
+
+    def reset(self, ctx: ExecutionContext):
+        ctx.ops.clear_view()
+        self.on_load(ctx)
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:622]: Calling `on_load()` directly is an anti-pattern I believe

File: plugins/panel-examples/__init__.py [hunk 8]
```diff
@@ -0,0 +1,940 @@
...
+            ),
+        )
+
+
+###
+# Dropdown Menu
+###
+
+
+class DropdownMenuPanel(foo.Panel):
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:651]: Can you update this to show how to build a menu like the embedding panel's that contains a cluster of dropdown + actions in a `menu()`? Extremely useful pattern

File: plugins/panel-examples/__init__.py [hunk 9]
```diff
@@ -0,0 +1,940 @@
...
+        )
+
+
+# Utility function to enhance styling of navigation buttons
+
+
+def add_panel_navigation(
+    panel, left=True, right=False, on_left=None, on_right=None
+):
+    base_btn_styles = {
```
Comment (brimoor) [inline plugins/panel-examples/__init__.py:899]: Should be updated to use `arrow_nav()` as done in https://github.com/voxel51/fiftyone/pull/4666

**Review by brimoor — DISMISSED** [review]
@ritch @imanjra can you please hop in here to help update these to make sure we're demonstrating best practices? I've left a possibly incomplete list of suggestions.

My goal is to finalize https://github.com/voxel51/fiftyone/pull/4637 by:
- Showing the essential elements of these as code blocks in the main docs
- Include an image/GIF in the main docs showing what the relevant component does/looks like
- Link to these examples for a fully runnable example

File: plugins/panel-examples/__init__.py [hunk 10]
```diff
@@ -0,0 +1,940 @@
...
+
+        ctx.panel.state.data = plot_data
+
+    def render(self, ctx: ExecutionContext):
+        panel = types.Object()
+
+        # grab data field from state and render it in a plotly view
+        panel.plot("data", label="Plotly Panel")  # shortcut for panel creation
+
+        panel.obj("data", view=types.PlotlyView())
```
Comment (ritch) [inline plugins/panel-examples/__init__.py:110]: dup (I'll fix)

File: plugins/panel-examples/__init__.py (hunk 8 above)
Comment (imanjra) [reply plugins/panel-examples/__init__.py:649]: I'll check this out

File: plugins/panel-examples/__init__.py (hunk 9 above)
Comment (imanjra) [reply plugins/panel-examples/__init__.py:897]: I'll check this out

File: plugins/panel-examples/__init__.py (hunk 4 above)
Comment (imanjra) [reply plugins/panel-examples/__init__.py:570]: How about `ctx.dataset.count_values("ground_truth.detections.label")`?

Comment (imanjra) [reply plugins/panel-examples/__init__.py]: Updated to use `count_values` for now

**Review by brimoor — APPROVED** [review]
LGTM

**Pull request merged**